src/Bicep.Core/Analyzers/Linter/LinterAnalyzer.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.Semantics;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+
+namespace Bicep.Core.Analyzers.Linter
+{
+    internal class LinterAnalyzer : IBicepAnalyzer
+    {
+        public static string AnalyzerName => "Bicep Internal Linter";
+        public static readonly bool IsCliInvoked;
+
+        private static readonly IEnumerable<Type> RuleTypes;
+        private static readonly ImmutableArray<IBicepAnalyzerRule> RuleSet;
+
+        static LinterAnalyzer()
```

Comment:
>static [](http://example.com/codeflow?start=8&length=6)

Btw, we've been trying to avoid as much static initialization as we can to avoid issues in tests.